include/RAJA/DepGraphNode.hxx

```diff
@@ -119,6 +108,33 @@ public:
   int& semaphoreReloadValue() { return m_semaphore_reload_value; }
 
   ///
+  /// Ready this task to be used again
+  ///
+  void reset() { m_semaphore_value.store(m_semaphore_reload_value); }
+
+  ///
+  /// Satisfy one incoming dependency
+  ///
+  void satisfyOne()
+  {
+    if (m_semaphore_value > 0) {
+      int val = --m_semaphore_value;
+    }
+  }
+
+  ///
+  /// Wait for all dependencies to be satisfied
+  ///
+  void wait()
+  {
+    while (m_semaphore_value > 0) {
+      // TODO: an efficient wait would be better here, but the standard
+      // promise/future is not good enough
+      std::this_thread::yield();
```

Comment:
This is going to be a key function point to optimize.   Unfortunately there is no best approach.  Do you want to add a comment for at least three possibilities to help people understand what the tradeoffs are?
(1) No waiting in the while loop will touch main memory 'too often', causing contention. (semaphore is volatile, which forces main memory touch).
(2) a small spin-loop on a volatile loop-control-variable is probably optimal,  but the 'correct' length for that loop is going to be kernel and system dependent.
(3) The yield overhead is O/S dependent, so you never know what delay you will get.  On some systems, the tasks may need to be very large for the yield overhead to not dominate execution time.